recipes/IWSLT22_lowresource/AST/transformer/train_speecht5.py

```diff
@@ -0,0 +1,365 @@
+#!/usr/bin/env python3
+"""Recipe for fine-tuning a SpeechT5 model for the ST task (no transcriptions).
+
+Author
+ * Haroun Elleuch, 2024
+"""
+
+import sys
+import torch
+import logging
+
+import speechbrain as sb
+
+from sacremoses import MosesDetokenizer
+from hyperpyyaml import load_hyperpyyaml
+from speechbrain.utils.distributed import run_on_main
+
+
+# Define training procedure
+class ST(sb.core.Brain):
+    def compute_forward(self, batch, stage):
+        """Forward computations from the waveform batches to the output probabilities."""
+
+        batch = batch.to(self.device)
+        wavs, wav_lens = batch.sig  # audio
+        bos_tokens, _ = batch.tokens_bos  # translation
+
+        encoder_output, logits, _ = self.modules.speecht5(
+            wav=wavs, decoder_input_ids=bos_tokens
+        )
+
+        log_probs = self.hparams.log_softmax(logits)
+
+        # compute outputs
+        hyps = None
+        if stage == sb.Stage.VALID:
+            # the output of the encoder is used for valid search
+            hyps, _, _, _ = self.hparams.valid_search(
+                encoder_output.detach(), wav_lens
+            )
+
+        elif stage == sb.Stage.TEST:
+            hyps, _, _, _ = self.hparams.test_search(
+                encoder_output.detach(), wav_lens
+            )
+
+        return log_probs, wav_lens, hyps
+
+    def compute_objectives(self, predictions, batch, stage):
+        """Computes the loss given predictions and targets."""
+        (log_probs, _, hyps) = predictions
+        ids = batch.id
+        tokens_eos, tokens_eos_lens = batch.tokens_eos
+
+        # st loss
+        loss = self.hparams.nll_loss(
+            log_probs, tokens_eos, length=tokens_eos_lens
+        )
+
+        if stage != sb.Stage.TRAIN:
+            fr_detokenizer = MosesDetokenizer(lang=self.hparams.lang)
+
+            if self.distributed_launch:
+                tokenizer = self.modules.speecht5.module.tokenizer
+            else:
+                tokenizer = self.modules.speecht5.tokenizer
```

Comment:
the reason why you are doing that is because you are not adding in your Brain object the tokenizer. Could you please delete these lines and add in your Brain object a tokenizer attribute such as :
```asr_brain.tokenizer = tokenizer``` ?